src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/StorageServices/NewAzureVhdVMCommand.cs

```diff
@@ -0,0 +1,576 @@
+                        DestinationPortRange = "8080"
+                    }
+                };
+            }
+
+            if (string.IsNullOrEmpty(this.DefaultVNetAddressSpace))
+            {
+                this.DefaultVNetAddressSpace = "10.0.0.0/16";
+            }
+
+            if (string.IsNullOrEmpty(this.DefaultSubnetAddressSpace))
+            {
+                this.DefaultSubnetAddressSpace = "10.0.0.0/24";
+            }
+        }
+
+        private void ValidateInitialInputParameters()
+        {
+            if (this.DiskFile == null && this.DiskLink == null)
+            {
+                throw new ArgumentException("DiskFile and DiskLink cannot be both null.");
+            }
+
+            if (this.DiskFile != null && this.DiskLink != null)
+            {
+                throw new ArgumentException("DiskFile and DiskLink cannot be both not null.");
+            }
+
+            if (this.DiskFile != null && this.DiskFile.Count() <= 0)
+            {
+                throw new ArgumentOutOfRangeException("DiskFile", "DiskFile input must contain at least one file.");
+            }
+
+            if (this.DiskLink != null && this.DiskLink.Count() <= 0)
+            {
+                throw new ArgumentOutOfRangeException("DiskLink", "DiskLink input must contain at least one link.");
+            }
+        }
+
+        private void CreateResourceGroupIfNotExists()
+        {
+            IResourceManagementClient resourceManagementClient = AzureSession.Instance.ClientFactory.CreateClient<ResourceManagementClient>(
+                        DefaultProfile.DefaultContext, AzureEnvironment.Endpoint.ResourceManager);
+            var resourceGroups = resourceManagementClient.ResourceGroups.List(new ResourceGroupListParameters { });
+            if (!resourceGroups.ResourceGroups.Any(r => string.Equals(r.Name, this.ResourceGroupName, StringComparison.OrdinalIgnoreCase)))
+            {
+                resourceManagementClient.ResourceGroups.CreateOrUpdate(this.ResourceGroupName, new ResourceGroup
+                {
+                    Location = this.Location
+                });
+            }
+        }
+
+        private void CreateDefaultStorageAccount()
+        {
+            InitializeStorageManagementClient();
+            string stoName = GetPrefixBasedResourceName("sto", false, true, 24);
+            stoAccount = this.storageClient.StorageAccounts.Create(this.ResourceGroupName, stoName, new StorageAccountCreateParameters
+            {
+                AccountType = AccountType.StandardGRS,
+                Location = this.Location
+            });
+            stoAccount = this.storageClient.StorageAccounts.GetProperties(this.ResourceGroupName, stoName);
+        }
+
+        private void InitializeStorageManagementClient()
+        {
+            if (this.storageClient == null)
+            {
+                this.storageClient = AzureSession.Instance.ClientFactory.CreateArmClient<StorageManagementClient>(
```

Comment:
In these places, we need to ensure that we are using the lowest version of StorageManagement and NetworkManagement possible - we should using the versions from Commands.Common.Network, and should create a similar common code library for storage as necessary.